backend/typescript/services/implementations/schedulingService.ts

```diff
@@ -708,12 +712,12 @@ class SchedulingService implements ISchedulingService {
         updatedScheduling.volunteer_id === null &&
         oldScheduling?.volunteer_id
```

Comment:
havent tested it out yet, but will be good to add another check `oldScheduling` then you can remove the question mark, and we know that oldScheduling def exists in that case

Reply:
I think the emails being sent in `updateSchedulingById` are for the case where volunteer cancels their own shift from PR #221 but I checked again by making change as a donor to a donation with an assigned volunteer and no cancellation emails were sent